lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -257,7 +256,15 @@ internal class EvaluatingCompiler(
      */
     fun eval(ast: ExprNode, session: EvaluationSession): ExprValue = compile(ast).eval(session)
 
+    /**
+     * Compiles the specified [ExprNode] into a [ThunkEnv].
+     *
+     * This operation is deeply recursive.
```

Comment:
Properly doing justice to the "why" here is beyond the scope of what I'm trying to accomplish here (this function was entirely undocumented before, it's not like I'm making the situation worse by leaving it mostly undocumented), but some placeholder documentation seemed in order since leaving the thread interruption behavior undocumented seemed unwise.